File: Courses/PHP Web Basics - май 2019/16.ExerciseEncapsulationInheritance/02.05.Animals/Tomcat.php
<?php

require_once "SoundProducible.php";
require_once "Animal.php";
require_once "Cat.php";

class Tomcat extends Cat
{
    const SOUND = "Give me one million b***h";
    const GENDER = "Male";

    /**
     * Kitten constructor.
     */
    public function __construct(string $name, int $age)
    {
        parent:: __construct($name, $age, self::GENDER);
    }

    public function ProduceSound()
    {
        return self::SOUND;
    }
}

File: Courses/C# OOP/Basics/10. Inheritance Exercise/Exercises_Inheritance/06.Animals/Animals/Tomcat.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _06.Animals.Animals
{
    public class Tomcat : Cat
    {
        private const string gender = "Male";
        public Tomcat(string name, int age) 
            : base(name, age, gender)
        {
        }
        public override void ProduceSound()
        {
            Console.WriteLine("MEOW");
        }
    }
}
